test/ddthost_test/test_reports.py

```diff
@@ -0,0 +1,197 @@
+#! /usr/bin/env python3
+"""
+-----------------------------------------------------------------------
+ Description:  Test capgen database report python interface
+
+ Assumptions:
+
+ Command line arguments: build_dir database_filepath
+
+ Usage: python test_reports <build_dir> <database_filepath>
+-----------------------------------------------------------------------
+"""
+import sys
+import os
+
+_TEST_DIR = os.path.dirname(os.path.abspath(__file__))
+_FRAMEWORK_DIR = os.path.abspath(os.path.join(_TEST_DIR, os.pardir, os.pardir))
+_SCRIPTS_DIR = os.path.join(_FRAMEWORK_DIR, "scripts")
+_SRC_DIR = os.path.join(_FRAMEWORK_DIR, "src")
+
+if not os.path.exists(_SCRIPTS_DIR):
+    raise ImportError("Cannot find scripts directory")
+# end if
+
+if ((sys.version_info[0] < 3) or
+    (sys.version_info[0] == 3) and (sys.version_info[1] < 8)):
+    raise Exception("Python 3.8 or greater required")
+# end if
+
+sys.path.append(_SCRIPTS_DIR)
+# pylint: disable=wrong-import-position
+from ccpp_datafile import datatable_report, DatatableReport
+# pylint: enable=wrong-import-position
+
+def usage(errmsg=None):
+    """Raise an exception with optional error message and usage message"""
+    emsg = "usage: {} <build_dir> <database_filepath>"
+    if errmsg:
+        emsg = errmsg + '\n' + emsg
+    # end if
+    raise ValueError(emsg.format(sys.argv[0]))
+
+if len(sys.argv) != 3:
+    usage()
+# end if
+
+_BUILD_DIR = os.path.abspath(sys.argv[1])
+_DATABASE = os.path.abspath(sys.argv[2])
+if not os.path.isdir(_BUILD_DIR):
+    _EMSG = "<build_dir> must be an existing build directory"
+    usage(_EMSG)
+# end if
+if (not os.path.exists(_DATABASE)) or (not os.path.isfile(_DATABASE)):
+    _EMSG = "<database_filepath> must be an existing CCPP database file"
+    usage(_EMSG)
+# end if
```

Comment:
```suggestion
import argparse

parser = argparse.ArgumentParser(description="Test capgen database report python interface")
parser.add_argument('build_dir')
parser.add_argument('database_filepath')

if len(sys.argv) > 3:
    parser.error("Too many arguments")

args = parser.parse_args()

_BUILD_DIR = os.path.abspath(args.build_dir)
_DATABASE = os.path.abspath(args.database_filepath)

if not os.path.isdir(_BUILD_DIR):
    parser.error("<build_dir> must be an existing build directory")

if (not os.path.exists(_DATABASE)) or (not os.path.isfile(_DATABASE)):
    parser.error("<database_filepath> must be an existing CCPP database file")
```
This version would allow us to add arguments later if needed more explicitly and we get help/usage output for free.

Usage is still printed to stderr like the current version with the custom error messages.

Reply:
argparsed!